File: datastructures/graph/dfs.py
class DFS:

    count = 0

    def __init__(self, g, s):
        self.marked = []
        for i in range(g.V):
            self.marked.append(False)
        self.dfs(g, s)

    def dfs(self, g, v):
        self.marked[v] = True
        self.count += 1
        for w in g.adj[v]:
            if not self.marked[w]:
                self.dfs(g, w)

File: datastructures/graph/dfs.go
package main

type DFS struct {
    marked []bool;
    count int;
}

func makeDFS(g *Graph, s int) *DFS {
    dfs := new(DFS)
    dfs.marked = make([]bool, g.V)
    dfs.dfs(g, s)
    return dfs
}

func (dfs *DFS) dfs(g *Graph, v int) {
    dfs.marked[v] = true
    dfs.count += 1
    for _, w := range g.adj[v] {
        if !dfs.marked[w] {
            dfs.dfs(g, w)
        }
    }
}
